go-app.js

```diff
@@ -232,14 +234,96 @@ go.app = function() {
     });
 
     self.states.add('Submit_Case', function(name) {
-      return new EndState(name, {
-        text: "Thank you! Your report has been submitted.",
-        next: 'Facility_Code_Entry'
-      });
+      return Q()
+        .then(function() {
+          // Send response to Ona
+          var ona_conf = self.im.config.ona;
+          if (typeof ona_conf == 'undefined') {
+            return self.im.log.info([
+              "No Ona API configured.",
+              "Not submitting data to Ona."
+            ].join(" "));
+          }
+          var ona = new Ona(self.im, {
+            auth: {
+              username: ona_conf.username,
+              password: ona_conf.password
+            },
+            url: ona_conf.url
+          });
+          var data = self.im.user.answers;
+          data.create_date_time = new Date();
+          data.reported_by = self.im.user.addr;
+          data.gender = (self.im.user.answers.No_SA_ID_Gender_Entry == 1) ? "male" : "female";
```

Comment:
I think there's a subtle bug here, the gender's state value is actually stored as a string not as an integer.
You can change the ChoiceState's `Choice` value to be `male` or `female`, that way you don't have to do this switch. The ChoiceState automatically maps the `1` & `2` entries to those values.